spawn_statestores/statestores_native/lib/statestores/adapters/native_snapshot_adapter.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Statestores.Adapters.NativeSnapshotAdapter do
+  @moduledoc """
+  Implements the behavior defined in `Statestores.Adapters.Behaviour` for Native databases using Mnesia.
+  """
+  use Statestores.Adapters.SnapshotBehaviour
+  use GenServer
+
+  require Logger
+
+  alias Statestores.Adapters.Native.SnapshotStore
+  alias Statestores.Schemas.Snapshot
+
+  @impl true
+  def get_by_key(id) do
+    :mnesia.dirty_index_read(SnapshotStore, id, :id)
+    |> case do
+      [] ->
+        nil
+
+      [data] ->
+        Snapshot.from_record_tuple(data)
+
+      error ->
+        Logger.error("Error getting snapshot by key: #{inspect(error)} #{inspect(id)}")
+        nil
+    end
+  end
+
+  @impl true
+  def get_by_key_and_revision(_id, _revision), do: raise("Not implemented")
```

Comment:
If we can implement the review it would be easier to adopt as the default persistence mechanism in Kubernetes